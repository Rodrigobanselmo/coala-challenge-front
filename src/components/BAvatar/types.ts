import { IUser } from "@/core/interfaces/IUser";

export interface IAvatarProps {
  name: string;
  photoUrl?: string | null;
}
