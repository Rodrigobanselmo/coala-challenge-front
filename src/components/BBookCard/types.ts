import { IBook } from "@/core/interfaces/IBook";
import { CircularProgressProps } from "@mui/material/CircularProgress";
import { IconButtonProps } from "@mui/material/IconButton";

export type BBookCardProps = {
  book: IBook;
  buttonLabel?: string;
  onClickButton?: (book: IBook) => Promise<void>;
};
