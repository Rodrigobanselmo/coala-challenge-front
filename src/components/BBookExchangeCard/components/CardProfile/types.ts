import { IUser } from "@/core/interfaces/IUser";

export interface IProfileProps {
  user?: IUser | null;
}
