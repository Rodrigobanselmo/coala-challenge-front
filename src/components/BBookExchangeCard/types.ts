import { IBook } from "@/core/interfaces/IBook";
import { IBookExchange } from "@/core/interfaces/IBookExchange";

export type BBookCardProps = {
  bookExchange: IBookExchange;
  buttonLabel?: string;
  onClickButton?: (book: IBookExchange) => Promise<void>;
};
