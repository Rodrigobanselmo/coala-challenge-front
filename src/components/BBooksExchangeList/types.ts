import { IBookExchange } from "@/core/interfaces/IBookExchange";
import { InfiniteData } from "@tanstack/react-query";
import { ReactNode } from "react";

export interface BBooksExchangeProps {
  data: InfiniteData<IBookExchange[]>;
  searchInputValue: string;
  onSearchInputChange: (value: string) => void;
  fetchNextPage: () => void;
  hasNextPage?: boolean;
  isFetchingNextPage: boolean;
  isLoading: boolean;
  loadMore?: boolean;
  buttonLabel?: string;
  onClickCardButton?: (book: IBookExchange) => Promise<void>;
  placeholder?: string;
  title: string;
  emptyBookComponent?: ReactNode;
  emptyBookLable?: string;
}
