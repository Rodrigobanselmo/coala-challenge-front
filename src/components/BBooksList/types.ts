import { IBook } from "@/core/interfaces/IBook";
import { IPaginationResult } from "@/core/interfaces/IPaginationResult";
import { InfiniteData } from "@tanstack/react-query";
import { ElementType, ReactNode } from "react";

export interface BBooksListProps {
  data: InfiniteData<IBook[]>;
  searchInputValue: string;
  onSearchInputChange: (value: string) => void;
  fetchNextPage: () => void;
  hasNextPage?: boolean;
  isFetchingNextPage: boolean;
  isLoading: boolean;
  loadMore?: boolean;
  buttonLabel?: string;
  onClickCardButton?: (book: IBook) => Promise<void>;
  placeholder?: string;
  emptyBookComponent?: ReactNode;
  emptyBookLable?: string;
  title: string;
}
