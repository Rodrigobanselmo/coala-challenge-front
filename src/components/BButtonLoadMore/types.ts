import { IBook } from "@/core/interfaces/IBook";
import { IPaginationResult } from "@/core/interfaces/IPaginationResult";
import { InfiniteData } from "@tanstack/react-query";
import { ElementType, ReactNode } from "react";

export interface ButtonLoadMoreListProps {
  fetchNextPage: () => void;
  canLoadMore: boolean;
  isFetchingNextPage: boolean;
}
