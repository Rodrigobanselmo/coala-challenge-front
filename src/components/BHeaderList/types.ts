import { IBook } from "@/core/interfaces/IBook";
import { IPaginationResult } from "@/core/interfaces/IPaginationResult";
import { InfiniteData } from "@tanstack/react-query";
import { ElementType, ReactNode } from "react";

export interface HeaderListProps {
  searchInputValue: string;
  onSearchInputChange: (value: string) => void;
  placeholder?: string;
  title: string;
  inputRef?: React.RefObject<HTMLInputElement>;
}
