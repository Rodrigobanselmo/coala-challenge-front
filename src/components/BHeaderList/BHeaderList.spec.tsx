import React from "react";
import { render } from "@testing-library/react";
import { HeaderList } from ".";
import { HeaderListProps } from "./types";

describe("HeaderList", () => {
  it("matches the snapshot", () => {
    const props: HeaderListProps = {
      searchInputValue: "Search Value",
      onSearchInputChange: () => {},
      placeholder: "Search Placeholder",
      title: "Header Title",
    };

    const { container } = render(<HeaderList {...props} />);
    expect(container).toMatchSnapshot();
  });
});
