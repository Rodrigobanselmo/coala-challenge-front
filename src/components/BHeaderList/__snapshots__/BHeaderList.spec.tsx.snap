// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderList matches the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1hfnxps"
  >
    <div
      class="MuiBox-root css-1bvhw10"
    >
      <div
        class="MuiBox-root css-70qvj9"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-pwpa85-MuiSvgIcon-root-MuiIcon-root"
          data-testid="AutoStoriesIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m19 1-5 5v11l5-4.5V1zM1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5V6c-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6zm22 13.5V6c-.6-.45-1.25-.75-2-1v13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5v2c1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5v-1.1z"
          />
        </svg>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1yqseqj-MuiTypography-root"
        >
          Header Title
        </h1>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-nl8vqx-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          placeholder="Search Placeholder"
          type="text"
          value="Search Value"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
